modules/ascat/main.nf

```diff
@@ -0,0 +1,155 @@
+process ASCAT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::ascat=3.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-c278c7398beb73294d78639a864352abef2931ce:dfe5aaa885de434adb2b490b68972c5840c6d761-0':
+        'quay.io/biocontainers/mulled-v2-c278c7398beb73294d78639a864352abef2931ce:dfe5aaa885de434adb2b490b68972c5840c6d761-0' }"
+
+    input:
+    tuple val(meta), path(normal_bam), path(normal_bai), path(tumor_bam), path(tumor_bai)
+    path(allele_files)
+    path(loci_files)
+
+    output:
+    tuple val(meta), path("*png"),               emit: png
+    tuple val(meta), path("*cnvs.txt"),          emit: cnvs
+    tuple val(meta), path("*purityploidy.txt"),  emit: purityploidy
+    tuple val(meta), path("*segments.txt"),      emit: segments
+    path "versions.yml",                         emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args           = task.ext.args        ?: ''
+    def prefix         = task.ext.prefix      ?: "${meta.id}"
+    def gender         = args.gender          ?  "$args.gender" :        "NULL"
+    def genomeVersion  = args.genomeVersion   ?  "$args.genomeVersion" : "hg19"
+    def purity         = args.purity          ?  "$args.purity" :        "NULL"
```

Comment:
Also adressing point [above](https://github.com/nf-core/modules/pull/1332#discussion_r823541913). 

Yes for the hg19 and good point. Shouldn't be hardcoded. I'll change to empty (and it'll break if not given, as it should).

But for the NULL things, it'll break the R-code that gets produced. Those expects explicitly `NULL` and in many cases handles that gracefully. But will break if it receives an empty string. 

One thing that could be done, is to re-code to a similar setup like the optional arguments ([five lines down](https://github.com/nf-core/modules/pull/1332/files#diff-09f30f18396b09d3f4f40bafc3f3c35e27bde3796ada26afa83cac729c3c50c8R34-R40)). It's because the R-functions wants named arguments (`function(something=value)`). So for the optional arguments, I make both the `something` and the `value` or else an empty string if user don't want to set that optional value.

Yeah, discusable, but really I think it's the way that best serves nextflow-adapt users, without pulling them too much through learning R-notation too.

Reply:
ok, sounds reasonable 👍